tests/PatternAssertions/TextFeaturedMediaAssert.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\PatternAssertions;
+
+use Symfony\Component\DomCrawler\Crawler;
+use PHPUnit\Framework\Exception;
+
+/**
+ * Assertions for text_featured_media pattern.
+ *
+ *  @see ./templates/patterns/text_featured_media/text_featured_media.ui_patterns.yml
+ */
+class TextFeaturedMediaAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions($variant): array {
+    return [
+      'title' => [
+        [$this, 'assertElementText'],
+        'div h2.ecl-u-type-heading-2.ecl-u-mt-2xl.ecl-u-mt-md-3xl.ecl-u-mb-l',
+      ],
+      'text' => [
+        [$this, 'assertElementText'],
+        'div .ecl-row div.ecl-col-md-6.ecl-editor',
+      ],
+      'caption' => [
+        [$this, 'assertElementText'],
+        'div.ecl-row figure figcaption.ecl-media-container__caption',
+      ],
+      'image' => [
+        [$this, 'assertImage'],
+        'div.ecl-row figure.ecl-media-container img',
+      ],
+      'video' => [
+        [$this, 'assertElementText'],
```

Comment:
adding iframe to the selector and then assertElementExists (or even assertElementAttribute for the src) would have made sense, but its ok for me like this since the pattern definition for video is basically free text

Reply:
Would be better avoid hardcode iframe here since video can use `<video>` tag.